### ISSUE TITLE
Code says done but nothing happens with UTM

### ISSUE THREAD

**Issue by Kinggtaon** [fritzlb/iOS17-JIT-WIN#31]
Code runs without any problems and it says done
But UTM under development certificate of Altstore says jit still needs to be activated.
I have a M2 ipad pro woth ios 17.3.1 and windows 11

**Comment by Kinggtaon** [comment]
Getting bundle ID...
DEBUG mode specified.
Got bundle ID: com.utmapp.UTM.HG
starting tunnel to device...
This might take a while. In case it freezes, either close this window and kill every python process in task manager or simply reboot your PC.
Sucessfully created tunnel: fdf2:5f30:56e6::1 64117
Manually trying to mount DeveloperDiskImage (this seems to prevent errors on some systems)...
2024-04-02 13:36:22 LAPTOP-QE4M04IT pymobiledevice3.cli.mounter[9080] ERROR DeveloperDiskImage already mounted

Diskimage already mounted.
Starting app...
(b'Process launched with pid 646\r\n', b'')
Started app. PID: 646
Starting debug server...
(b"\r\nFollow the following connections steps from LLDB:\r\n\r\n(lldb) platform select remote-ios\r\n(lldb) target create /path/to/local/application.app\r\n(lldb) script lldb.target.module[0].SetPlatformFileSpec(lldb.SBFileSpec('/private/var/containers/Bundle/Application/<APP-UUID>/application.app'))\r\n(lldb) process connect connect://[fdf2:5f30:56e6::1]:55841   <-- ACTUAL CONNECTION DETAILS!\r\n(lldb) process launch\r\n\r\n", b'')
Started debug server with connection details: [fdf2:5f30:56e6::1]:55841
Run debugging commands...
This might take a few minutes.
(lldb) command source -s 0 cmdfile.txt
Executing commands in 'C:\Users\Abbas\Desktop\IOS17-~1\cmdfile.txt'.
(lldb) gdb-remote [fdf2:5f30:56e6::1]:55841
(lldb) settings set target.memory-module-load-level minimal
(lldb) attach -p 646
Process 646 stopped
* thread #1, stop reason = signal SIGSTOP
    frame #0: 0x00000001d9491178 libsystem_kernel.dylib
->  0x1d9491178: ret
    0x1d949117c: mov    x16, #-0x30               ; =-48
    0x1d9491180: svc    #0x80
    0x1d9491184: ret
Executable module set to "\private\var\containers\Bundle\Application\46922585-84E0-4705-89CF-58105A06954E\UTM.app\UTM".
(lldb) detach
Process 646 detached
(lldb) (lldb)  quit
 Could not find implementation lookup function "class_getMethodImplementation" step in through ObjC method dispatch will not work.
warning: libobjc.A.dylib is being read from process memory. This indicates that LLDB could not find the on-disk shared cache for this device. This will likely reduce debugging performance.

(lldb) warning: Architecture changed from arm64e-apple-ios to arm64-apple-ios.

done.
DEBUG mode specified. This script won't quit automatically. Press CTRL + C to stop it.

**Comment by fritzlb** [comment]
Interesting, it seems like UTM (or a part of the app, idk) crashes when you run the script. That's kinda weird and shouldn't happen. How did you install UTM? If I interpret the log correctly (and if you didn't modify it to not include the last characters of your bundle ID) it probably wasn't altstore in combination with a free developer account

**Comment by Kinggtaon** [comment]
yes, I have installed UTM using Altstore and it gave me the exact error. My Altstore is expired now and I cant install UTM with it for a weak so i had to use Sideloadly to get you the debug output. Installing Altstore on Ios 17 requires MacOS and i dont have it currently.

**Comment by fritzlb** [comment]
AltStore should work with iOS 17, even on windows. I can test this maybe in a few minutes again. 
I personally own a m2 device too, and for me UTM (the one requiring JIT, I didn’t test the other) worked. Again, I‘ll report if something’s changed by now. 
oh and thank you for the debug messages :)

**Comment by fritzlb** [comment]
Okay, Altstore still works.
UTM, at least for me, kinda works too, if the RAM is set to something reasonnable, like 1GB for Mx-Devices. Also I only managed to get pretty lightweight linux distros to run, debian 11 with LXDE for example. Oh and Debian 12 boots but isn't really usable.
I get those presumably crash messages too, probably thats UTM related and can be ignored for now.

If you really need a windows soloution I stumbled across this https://github.com/nythepegasus/SideJITServer/releases repo, I didn't test it though because I don't really need JIT on iOS that often (if then only to play MC and for that it works fine).

In general I'm recently quite disappointed with Apples behaviour towards devs, especially regarding the DMA, so I don't spend that much time developing anything iOS related anymore, still it would be cool for sth like JitStreamer to be compatible with iOS 17.

Still I'm glad now JIT works for you even if it's by using a mac.
I'll close this for now, but feel free to reopen if there's anything not working again :)
